### PR TITLE
Minimum stable dependecies, random_bytes() in user token generator, new Travis matrix

### PR DESCRIPTION
**Pull request by szymach**
This change is required in order for anyone using `master` branch to be able to update their dependencies. So after this gets merged the `remove-fosub` branch will be rebased.

### CODE REVIEW

**Comment by chives** [comment]
We can use `random_bytes` and https://github.com/paragonie/random_compat instead of Symfony\Component\Security\Core\Util\SecureRandomInterface (as symfony itself does) in order to allow newest `symfony/security` versions (> 2.8)

File: Controller/AdminController.php [hunk 1]
```diff
@@ -18,12 +18,11 @@
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
-use Symfony\Component\Security\Core\SecurityContext;
 
 class AdminController
 {
     /**
-     * @var \Symfony\Component\Security\Core\SecurityContext
```
Comment (szymach) [inline Controller/AdminController.php]: since `TokenStorageInterface` is being used anyway

File: Token/TokenGenerator.php [hunk 2]
```diff
@@ -9,25 +9,13 @@
...
-        $this->secureRandom = $secureRandom;
-    }
-
-    /**
      * @return string
      */
     public function generateToken()
     {
-        return str_replace(['+', '/', '='], '', base64_encode($this->secureRandom->nextBytes(32)));
+        return rtrim(strtr(base64_encode(32), '+/', '-_'), '=');
```
Comment (szymach) [inline Token/TokenGenerator.php]: copied from [Symfony's UriSafeTokenGenerator.php](https://github.com/symfony/security/blob/1d2585926ab63fd5b139b1f86111dd1b2cbccb1b/Csrf/TokenGenerator/UriSafeTokenGenerator.php#L50)

File: composer.json [hunk 3]
```diff
@@ -14,10 +14,12 @@
         "php": ">=5.4.0",
         "fsi/admin-bundle" : "~1.1@dev",
         "symfony/framework-bundle": "~2.3",
+        "symfony/security": "~2.3",
```
Comment (szymach) [inline composer.json]: `security-bundle` is `~2.6` does it matter?

File: Token/TokenGenerator.php (hunk 2 above)
Comment (rn0) [inline Token/TokenGenerator.php]: this is invalid :) `base64_encode(32)` will return same string every time.

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by norzechowicz** [comment]
test on php 5.4 with --prefer-lowest flag are failing

**Comment by szymach** [comment]
@fsi-open tests run fine now

**Comment by chives** [comment]
:+1:

**Comment by norzechowicz** [comment]
if you are chaning dependencies anyway I think it would be better to use ``^`` operator instead of ``~``. At least according to composer documentation.

**Comment by szymach** [comment]
@fsi-open changed to caret

**Comment by JarekW** [comment]
:+1:

File: composer.json [hunk 4]
```diff
@@ -13,30 +13,35 @@
     "require": {
         "php": ">=5.4.0",
         "fsi/admin-bundle" : "~1.1@dev",
```
Comment (norzechowicz) [inline composer.json]: Same thing as here: https://github.com/fsi-open/admin-positionable-bundle/pull/21#discussion_r51563632

**Pull request merged**